accounts-service/com.endlessm.ParentalControls.rules

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright © 2019 Endless Mobile, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * Authors:
+ *  - Philip Withnall <withnall@endlessm.com>
+ */
+
+polkit.addRule(function(action, subject) {
+    /* Allow administrators to read parental controls (for any account) without
+     * needing an additional polkit authorisation dialogue. Also allow them to
+     * change their own (but not others’) for the same reason. */
```

Comment:
I'm not so sure about the last bit. We don't allow administrators to change any of the other settings for other users without first clicking Unlock; and once they've clicked Unlock, approval's not needed to make further changes (for a while?). Why should this be different?

Reply:
‘the last bit’ being allowing admins to change their own parental controls settings? I think it’s a bit of a moot point really: as per https://phabricator.endlessm.com/T23897#666377, I don’t think we should be showing/sensitising the parental controls widgets for admins, since the whole concept of applying parental controls to an admin account doesn’t make sense.

Happy to drop that bit of the change though. Will do so shortly.

Reply:
Ah, I think I misunderstood this change. I think the admin should have to click Unlock to be able to modify other users' parental controls, complete a polkit challenge, and then not have to complete any further polkit challenges. Not sure about the user changing their own. Agreed it doesn't make much sense!

Reply:
I agree with you about wanting a single polkit challenge when clicking the ‘Unlock’ button, and no further challenges. That’s partially solved by this (which allows reading the data without a challenge, before the ‘Unlock’ button is clicked), and partially solved by another PR I’m working on now, which makes the controls insensitive while the user doesn’t have authorisation (i.e. before the ‘Unlock’ button is clicked, or after the temporary authorisation times out in the session).

So I think this is ready to merge, if Jenkins is going to cooperate.